Lib/asyncio/queues.py

```diff
@@ -210,9 +272,58 @@ async def join(self):
         queue. The count goes down whenever a consumer calls task_done() to
         indicate that the item was retrieved and all work on it is complete.
         When the count of unfinished tasks drops to zero, join() unblocks.
+
+        Raise a QueueShutDown if _shutdown_state is set to
+        `SHUTDOWN_IMMEDIATE`.
```

Comment:
```suggestion
        Raises QueueShutDown if the queue has been shut down immediately.
```

Doc-strings are for humans